### PR TITLE
Updated clean developer doc for class based stages

### PR DESCRIPTION
**Pull request by danielgraziano**
to view the made doc:
1. clone repo
2. checkout this branch
3. cd into docs folder
4. make docs
5. view the made docs

### CODE REVIEW

File: docs/clean/developer_guide/stages/location.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    * | OS-specific Clean Stages reside in the `stages/<os>/stages.py` file.
+      | For example, the IOSXE specific Clean Stages belong `here <https://github.com/CiscoTestAutomation/genielibs/blob/master/pkgs/clean-pkg/src/genie/libs/clean/stages/iosxe/stages.py>`_.
+
+    * | Platform-specific Clean Stages reside in the `stages/<os>/<platform>/stages.py` file.
+      | For example, the IOSXE CAT9K Clean Stages belong `here <https://github.com/CiscoTestAutomation/genielibs/blob/master/pkgs/clean-pkg/src/genie/libs/clean/stages/iosxe/cat9k/stages.py>`_.
+
+.. note::
+
+    If the directory structure and/or the `stages.py` file does not already exist, create it following the below example but use your `<os>` and `<platform>`.
```
Comment (tahigash) [inline docs/clean/developer_guide/stages/location.rst:17]: better to tell os/platform need to aligh with what unicon.plugins has.

File: docs/clean/developer_guide/stages/steps.rst [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                step.passed("Successfully deleted configured boot variables")
+
+        def configure_boot_variable(self, steps, device, images, timeout=TIMEOUT,
+                                    current_running_image=CURRENT_RUNNING_IMAGE):
+
+            with steps.start("Set boot variable to images provided for {}".format(
+                    device.name)) as step:
+
+                if current_running_image:
+                    log.info("Retrieving and using the running image due to "
```
Comment (tahigash) [inline docs/clean/developer_guide/stages/steps.rst:179]: from here, user might copy and paste. I understand not main point. but might be better to have `import logging` and another line which define `log` at top.

File: docs/clean/developer_guide/stages/execution_order.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+1. delete_boot_variable
+2. configure_boot_variable
+3. set_configuration_register
+4. write_memory
+5. verify_boot_variable
+6. verify_configuration_register
+
+.. note::
+
+    This documentation did not show the development of the remaining steps, though, they have been added for completeness.
```
Comment (tahigash) [inline docs/clean/developer_guide/stages/execution_order.rst:17]: might be better tell like "not mandatory, but writing method in order as same as `exec_order` as best practice."

File: docs/clean/developer_guide/stages/abstracted.rst [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        # Execution order of Stage steps
+        # ==============================
+        exec_order = [
+            'delete_boot_variable',
+            'configure_boot_variable',
+            'write_memory',
+            'verify_boot_variable'
+        ]
+
+If we needed to overwrite delete_boot_variable to change the behaviour all we do is define the method again and write our new logic.
```
Comment (tahigash) [inline docs/clean/developer_guide/stages/abstracted.rst:233]:
```suggestion
If we needed to overwrite `delete_boot_variable` to change the behaviour all we do is define the method again and write our new logic.
```

**Review by tahigash — APPROVED** [review]
approved with small comments.

I believe not high demand and no rush. but please take AI for development guide of device discovery.

File: docs/clean/developer_guide/stages/steps.rst [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                step.passed("Successfully deleted configured boot variables")
+
+Passing Step Arguments
+----------------------
+
+To pass arguments to a clean step, add the arguments to the method and if there's a default, set it equal to that. Here's an example were we pass `images`, `timeout`, and `current_running_image`. You can see we provided the defaults by using:
+
+`timeout=TIMEOUT` `current_running_image=CURRENT_RUNNING_IMAGE`
+
+During runtime these values will be overwritten with user provided (if provided). You can then use these arguments like you would in any other python method.
```
Comment (LukasMcClelland) [inline docs/clean/developer_guide/stages/steps.rst:125]: Word missing I think

```suggestion
During runtime these values will be overwritten with user provided values (if provided). You can then use these arguments like you would in any other Python method.
```

**Review by LukasMcClelland — APPROVED** [review]
Well written!

**Pull request merged**